### PR TITLE
Fix:Update starlette.asciidoc, incorrect middleware order

### PR DESCRIPTION
**Pull request by mariocandela**
## What does this pull request do?

The documentation contains an error, you need to put the APM middleware before BaseHTTPMiddleware.

### CODE REVIEW

**Comment by mariocandela** [comment]
> I don't think this is correct:
> 
> > Using BaseHTTPMiddleware will prevent changes to [contextlib.ContextVar](https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar)s from propagating upwards.
> 
> So, BaseHTTPMiddleware needs to be above (i.e. before) our middleware in the list so that we have an unbroken contextvar chain from the endpoint to our middleware.
> 
> Does this not match your experience?
> 
> Edit: see also the original issue: #1701
Yesterday, we integrated the APM agent across all our microservices, placing the agent's middleware as the last element. No data was reaching the APM server in this configuration. However, when we placed it as the first element, everything worked correctly. Could it be that one of our custom middlewares was invalidating the context variable?

Thanks for your time Colton :1st_place_medal:

**Comment by basepi** [comment]
Are you passing in the middleware as a list or using the `add_middleware()` function repeatedly? I think that may be where the confusion is -- the original issue #1701 was using `add_middleware`, which appears to insert the middleware at the front of the list. Thus, the required order would be swapped based on how you're adding the middleware.

I think we need to expand this warning with examples of both.

**Comment by v1v** [comment]
run docs-build

**Comment by basepi** [comment]
@mariocandela Can you answer my question above?

**Comment by mariocandela** [comment]
> @mariocandela Can you answer my question above?

Hi @basepi, sorry for the delay, we use middleware as a list.
Initially, we had the agent's middleware placed last, but data wasn't reaching the APM server in this setup. However, when we moved it to the first position, everything started working correctly.

I haven't had the chance to delve deeper into the issue yet. It's possible that our custom middleware, which utilizes dependency injection, could be causing the problem. I'll make sure to allocate some time to debug the APM and figure out what's going on with the contextvar.

Thanks for your time

**Comment by basepi** [comment]
Alright, you've confirmed my suspicion. Our middleware must be first in the list if you're passing in a list of middleware, but last if you're adding each one manually with add_middleware. I'll add a commit to your PR clarifying that point.

File: docs/starlette.asciidoc [hunk 1]
```diff
@@ -43,7 +43,7 @@ app.add_middleware(ElasticAPM)
 ----
 
 WARNING: If you are using any `BaseHTTPMiddleware` middleware, you must add them
-*before* the ElasticAPM middleware. This is because `BaseHTTPMiddleware` breaks
+*after* the ElasticAPM middleware. This is because `BaseHTTPMiddleware` breaks
 `contextvar` propagation, as noted
 https://www.starlette.io/middleware/#limitations[here].
```
Comment (basepi) [inline docs/starlette.asciidoc:48]:
```suggestion
WARNING: `BaseHTTPMiddleware` breaks `contextvar` propagation, as noted
https://www.starlette.io/middleware/#limitations[here]. This means the
ElasticAPM middleware must be above any `BaseHTTPMiddleware` in the final
middleware list. If you're calling `add_middleware` repeatedly, add the
ElasticAPM middleware last. If you're passing in a list of middleware,
ElasticAPM should be first on that list.
```

**Review by basepi — CHANGES_REQUESTED** [review]
One change. Thanks again for talking through this with me!

**Comment by mariocandela** [comment]
> One change. Thanks again for talking through this with me!

thank you for your time!

**Review by basepi — APPROVED** [review]

**Comment by basepi** [comment]
run docs-build

**Pull request merged**